devito/arguments.py

```diff
@@ -175,51 +212,41 @@ def dtype(self):
 
     @cached_property
     def rtargs(self):
-        if self.size is not None:
-            return []
-        else:
-            size = ScalarArgument("%s_size" % self.name, self, max)
-            return [size]
-
-    @property
-    def decl(self):
-        return self.rtargs[0].decl
+        size = ScalarArgument("%s_size" % self.name, self, max)
+        return [size]
 
     # TODO: Can we do without a verify on a dimension?
     def verify(self, value):
         verify = True
+        if value is None:
+            if self.value is not None:
+                return True
 
-        if value is None and self._value is not None:
-            return verify
+            try:
```

Comment:
This try-catch is probably a symptom that it'd be better to have a ``BufferedDimensionArgProvider``. If that's the case, then we could also have a superclass for these DimensionArgProvider classes

Reply:
I agree but can we consider that outside the scope of this PR?

Reply:
Alright